src/DotNet/Importer.cs

```diff
@@ -650,7 +650,7 @@ public Importer(ModuleDef module, ImporterOptions options, GenericParamContext g
 			try {
 				// Get the original field def in case the declaring type is a generic
 				// type instance and the field uses a generic type parameter.
-				origField = fieldInfo.Module.ResolveField(fieldInfo.MetadataToken);
+				origField = fieldInfo.DeclaringType is { IsGenericType: true } ? fieldInfo.Module.ResolveField(fieldInfo.MetadataToken) : fieldInfo;
```

Comment:
Is this code exactly identical to the original code?

Reply:
I think so, `typeof(C1<>.C2).IsGenericType` is `true`, so if `IsGenericType` is `false`, I don't think `ResolveField` can return a different value
```C#
public class C1<T>
{
    public class C2
    {
        public T Field;
    }
}
```

Reply:
I can only accept this PR if it's guaranteed to work the same as before.